compiler/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -155,13 +155,23 @@ impl CrateDefMap {
 
     /// Go through all modules in this crate, find all `contract ... { ... }` declarations,
     /// and collect them all into a Vec.
-    pub fn get_all_contracts(&self) -> Vec<Contract> {
+    pub fn get_all_contracts(&self, interner: &NodeInterner) -> Vec<Contract> {
         self.modules
             .iter()
             .filter_map(|(id, module)| {
                 if module.is_contract {
-                    let functions =
+                    let function_ids: Vec<FuncId> =
                         module.value_definitions().filter_map(|id| id.as_function()).collect();
+
+                    let functions = function_ids
+                        .into_iter()
+                        .map(|id| {
```

Comment:
We shouldn't `collect` into `function_ids` then call `into_iter` immediately afterward, the intermediate `Vec` is not needed.